bin/cli-flags.js

```diff
@@ -21,21 +21,25 @@ module.exports = {
       negative: true,
     },
     {
-      name: 'profile',
+      name: 'client-progress',
       type: Boolean,
-      describe: 'Print compilation profile data for progress steps',
-    },
-    {
-      name: 'progress',
-      type: Boolean,
-      describe: 'Print compilation progress in percentage',
+      describe: 'Print compilation progress in percentage in the browser',
       group: BASIC_GROUP,
+      processor(opts) {
+        opts.client = opts.client || {};
+        opts.client.progress = opts.clientProgress;
+        delete opts.clientProgress;
+      },
```

Comment:
hm, I think webpack serve should do it automatically, where is the problem?

Reply:
Oh, I see, good name, I have ideas for refactoring here, but let's do it late

Reply:
Currently, argument processing is done manually in https://github.com/webpack/webpack-cli/blob/master/packages/serve/src/createConfig.ts and commander doesn't support nested properties. For better manageability, I've opened https://github.com/webpack/webpack-cli/pull/2133 to enable `cli-flags` to process them.